How to disable `rule.exported` for consts?

## TL;DR:

I am aware that the goal of this project is no longer to be a 100% drop-in replacement for golint (which I think is fair) but is there a way for me to disable the `exported` rule but only for consts?

## Full Details

At [pipeline-components](https://pipeline-components.dev/) we have a component for [go-lint](https://gitlab.com/pipeline-components/go-lint). Although it is not one of our larger components<sup>\[1]</sup>, I'm thinking of prolonging support by replacing `golint` with revive.

However, the current claim that the default behaviour is the same as `golint` does not appear to be 100% accurate.

To be specific, I am seeing the following warnings from revive that are not there with golint:

1. "should have a package comment"
2. "this block is empty, you can remove it"
3. "comment on exported const [..] should be of the form [..]"

The first two I can resolve by tweaking the default config,<sup>\[2]</sup>, which is fine for our purposes (as golint does not have a config, so we can just add one for revive).

But I have not yet been able to resolve the `comment on exported const [..] should be of the form [..]`. Please note, this is explicitly about exported "const", not other exported types.

How could this be achieved?

---

<sup>\[1]</sup> Currently it only has +/- 40K pulls on DockerHub ![][1]<br>
<details><summary><sup>[2]</sup> Click to see config</summary>

```toml
ignoreGeneratedHeader = false
severity = "warning"
confidence = 0.8
errorCode = 0
warningCode = 0

[rule.blank-imports]
[rule.context-as-argument]
[rule.context-keys-type]
[rule.dot-imports]
[rule.empty-block]
    Disabled = true
[rule.errorf]
[rule.error-naming]
[rule.error-return]
[rule.error-strings]
[rule.exported]
[rule.increment-decrement]
[rule.indent-error-flow]
[rule.package-comments]
     Disabled = true
[rule.range]
[rule.receiver-naming]
[rule.redefines-builtin-id]
[rule.superfluous-else]
[rule.time-naming]
[rule.unexported-return]
[rule.unreachable-code]
[rule.unused-parameter]
[rule.var-declaration]
[rule.var-naming]
```

</details>


[1]: https://img.shields.io/docker/pulls/pipelinecomponents/go-lint.png

Hi @Potherca, I'm not sure to understand your request on linting comments on public constants.
For the file `foo.go` 
```go
package main

// this is not ok
const AConstant = 10
```

When linting it with [`golint`](https://github.com/golang/lint/tree/master) I get:
```
$ golint .
foo.go:3:1: comment on exported const AConstant should be of the form "AConstant ..."
```
and when linting with `revive` I get the same failure:
```
$ revive .
foo.go:3:1: comment on exported const AConstant should be of the form "AConstant ..."
```
Thus it seems to me that for the case of public constants comments, `revive` is iso-functional to `golint`. I'm missing something?

@chavacava Thank you for taking the time to look at this.

To reproduce what I am seeing:

```go
package main

const (
	// this is not ok
	AConstant = 10

	Bconstant = 11 // Some comment here
)
```

Running revive and golint side-by-side[^1] shows:

[^1]: Using `diff <(~/go/bin/revive ./... | sort) <(docker run --rm -v "${PWD}:/code/" pipelinecomponents/go-lint ./... | sort)`

```diff
1,2d0
< 
< main.go:1:1: should have a package comment
4c2
< main.go:7:17: comment on exported const Bconstant should be of the form "Bconstant ..."
---
> main.go:7:2: exported const Bconstant should have comment (or a comment on this block) or be unexported
```

So it looks like the comment behind the const is the cause of the difference between golint and revive...

--- a/rule/exported.go
+++ b/rule/exported.go
@@ -204,8 +204,8 @@ func (w *lintExported) lintTypeDoc(t *ast.TypeSpec, doc *ast.CommentGroup) {
 		}
 	}
 	// if comment starts with name of type and has some text after - it's ok
-	expectedPrefix := t.Name.Name+" "
-	if strings.HasPrefix(s, expectedPrefix){
+	expectedPrefix := t.Name.Name + " "
+	if strings.HasPrefix(s, expectedPrefix) {
 		return
 	}
 	w.onFailure(lint.Failure{
@@ -244,7 +244,7 @@ func (w *lintExported) lintValueSpecDoc(vs *ast.ValueSpec, gd *ast.GenDecl, genD
 		return
 	}
 
-	if vs.Doc == nil && vs.Comment == nil && gd.Doc == nil {
+	if vs.Doc == nil && gd.Doc == nil {
 		if genDeclMissingComments[gd] {
 			return
 		}
@@ -348,7 +348,7 @@ func (w *lintExported) doCheckPublicInterface(typeName string, iface *ast.Interf
 
 func (w *lintExported) lintInterfaceMethod(typeName string, m *ast.Field) {
 	if len(m.Names) == 0 {
-		return 
+		return
 	}
 	if !ast.IsExported(m.Names[0].Name) {
 		return

--- a/testdata/golint/const-block.go
+++ b/testdata/golint/const-block.go
@@ -4,16 +4,16 @@
 package foo
 
 const (
-	InlineComment = "ShouldBeOK" // InlineComment is a valid comment
+	InlineComment = "ShouldBeOK" // InlineComment is not a valid documentation
+	// MATCH:7 /exported const InlineComment should have comment (or a comment on this block) or be unexported/
 
-	// Prefix for something.
 	// MATCH /comment on exported const InlineWhatever should be of the form "InlineWhatever ..."/
 	InlineWhatever = "blah"
 
 	Whatsit = "missing_comment"
-	// MATCH:13 /exported const Whatsit should have comment (or a comment on this block) or be unexported/
 
 	// We should only warn once per block for missing comments,
+	// thus do not warn on:13 /exported const Whatsit should have comment (or a comment on this block) or be unexported/
 	// but always complain about malformed comments.
 
 	WhosYourDaddy = "another_missing_one"

